### DISCUSSION TITLE
Investigate survival model alternatives to existing lifetime models

### DISCUSSION THREAD

**Discussion by ricardoV94** [pymc-labs/pymc-marketing#1231] [Ideas]
Could be more flexible:
  1. Easier to integrate covariates
  2. Easier to customize (priors, likelihoods, arbitrary structure)
  3. More similar approach to MMM (have on robust model that can be customized)

**Comment by larryshamalama** [comment]
> 1. Easier to integrate covariates

Do you mean _incorporate_? This is very nit-picky for now, but integration of covariates within longitudinal (especially causal) has a very specific meaning (c.f. G-Formula or G-Computation).

My understanding is that the bigger goal for CLVs is to relax our model assumptions. In biostats, these purchase times can be referred as recurrent events or, when an outcome (e.g. purchase amounts) at these times are of primary interest, irregularly observed data or even covariate-driven/outcome-dependent/etc. observation times. I like [Pullenayegum and Lim (2016)](https://pubmed.ncbi.nlm.nih.gov/24855119/)'s review paper and [Ryu et al. (2007)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2288578/)'s paper, the latter is Bayesian.

Modelling of the outcome and timings can be done via joint modelling, i.e. having dependence between models parameters. All of this can also have sequential interventions, which can add causal layer to complicate things further (or make things more interesting).

This opens up sort of a rabbit hole in longitudinal modelling that I particularly enjoy, hence my bias.

**Comment by ricardoV94** [comment]
ya incorporate!

**Comment by ColtAllen** [comment]
I'm planning to work on https://github.com/pymc-labs/pymc-marketing/issues/134 soon. The methodology for adding time-invariant covariates is rooted in survival analysis:

https://www.brucehardie.com/notes/019/time_invariant_covariates.pdf
